src/test/java/uk/gov/pay/directdebit/webhook/gocardless/services/SendEmailsForGoCardlessPaymentEventsHandlerTest.java

```diff
@@ -17,58 +18,63 @@
+public class SendEmailsForGoCardlessPaymentEventsHandlerTest {
 
     @Mock
     private UserNotificationService mockUserNotificationService;
 
     @Mock
     private PaymentQueryService mockPaymentQueryService;
-
+    
     @InjectMocks
-    private GoCardlessPaymentHandler goCardlessPaymentHandler;
+    private SendEmailsForGoCardlessEventsHandler sendEmailsForGoCardlessEventsHandler;
 
     private GoCardlessOrganisationId organisationIdentifier = GoCardlessOrganisationId.valueOf("test_org_id");
     private GatewayAccountFixture gatewayAccountFixture = GatewayAccountFixture.aGatewayAccountFixture().withOrganisation(organisationIdentifier);
     private MandateFixture mandateFixture = MandateFixture.aMandateFixture().withGatewayAccountFixture(gatewayAccountFixture);
     private Payment payment = PaymentFixture.aPaymentFixture().withMandateFixture(mandateFixture).toEntity();
-    private GoCardlessEventFixture goCardlessEventFixture = GoCardlessEventFixture.aGoCardlessEventFixture().withLinksOrganisation(organisationIdentifier);
+    private GoCardlessEventFixture goCardlessEventFixture = GoCardlessEventFixture.aGoCardlessEventFixture()
+            .withResourceType(GoCardlessResourceType.PAYMENTS)
+            .withLinksOrganisation(organisationIdentifier);
 
     @Rule
     public ExpectedException thrown = ExpectedException.none();
 
     @Test
-    public void handle_onFailedPaymentGoCardlessEvent_shouldSendEmail() {
-        GoCardlessEvent goCardlessEvent = spy(goCardlessEventFixture.withAction("failed").toEntity());
+    public void assertEmailIsSentForFailedPaymentGoCardlessEvent() {
+        GoCardlessEvent goCardlessEvent = goCardlessEventFixture.withAction("failed").toEntity();
 
         when(mockPaymentQueryService.findByGoCardlessPaymentIdAndOrganisationId(goCardlessEvent.getLinksPayment().get(),
                 goCardlessEvent.getLinksOrganisation()))
                 .thenReturn(Optional.of(payment));
 
-        goCardlessPaymentHandler.handle(goCardlessEvent);
+        sendEmailsForGoCardlessEventsHandler.handle(List.of(goCardlessEvent));
 
         verify(mockUserNotificationService).sendPaymentFailedEmailFor(payment);
     }
 
     @Test
-    public void handle_onCreatePaymentGoCardlessEvent_shouldThrowExceptionWhenEventHasNoLinkedPayment() {
+    public void assertEmailIsSentForCreatePaymentGoCardlessEvent_shouldThrowExceptionWhenEventHasNoLinkedPayment() {
         GoCardlessEvent goCardlessEvent = goCardlessEventFixture
                 .withLinksOrganisation(GoCardlessOrganisationId.valueOf("does_not_exist"))
                 .withAction("created")
                 .withLinksPayment(null)
                 .toEntity();
 
         thrown.expect(GoCardlessEventHasNoPaymentIdException.class);
-        goCardlessPaymentHandler.handle(goCardlessEvent);
+        sendEmailsForGoCardlessEventsHandler.handle(List.of(goCardlessEvent));
+        verify(mockUserNotificationService, never()).sendPaymentFailedEmailFor(any());
```

Comment:
Do we have any tests that check we handle multiple events at once?

Reply:
i've amended assertEmailIsSentForFailedPaymentGoCardlessEvent to deal with two events instead of one